fenix/app/src/androidTest/java/org/mozilla/fenix/helpers/TestHelper.kt

```diff
@@ -339,29 +338,25 @@ object TestHelper {
         }
     }
 
+    // Method used to press allow on the different system permission dialogs
     fun grantPermission() {
         if (Build.VERSION.SDK_INT >= 23) {
-            when (Build.VERSION.SDK_INT) {
-                Build.VERSION_CODES.R ->
-                    itemWithResIdAndText(
-                        "com.android.permissioncontroller:id/permission_allow_foreground_only_button",
-                        "While using the app",
-                    ).also {
-                        it.waitForExists(waitingTime)
-                        it.click()
-                    }
-                else ->
-                    itemWithResIdAndText(
-                        "com.android.packageinstaller:id/permission_allow_button",
-                        "ALLOW",
-                    ).also {
-                        it.waitForExists(waitingTime)
-                        it.click()
-                    }
+            if (mDevice.findObject(UiSelector().textContains("While using the app")).waitForExists(
+                    waitingTimeShort,
+                )
+            ) {
+                mDevice.findObject(UiSelector().textContains("While using the app")).click()
+            } else {
+                mDevice.findObject(
+                    UiSelector()
+                        .textContains("Allow")
+                        .className("android.widget.Button"),
+                ).click()
```

Comment:
Is the `BUILD.VERSION_CODES` check not needed anymore?

Reply:
No, the storage access prompt is different than the others on the same API, so the API check is not relevant. Instead, I thought to do this instead: if there's a different button than "Allow", click it, otherwise click "Allow".